pkg/sql/insert.go

```diff
@@ -223,6 +226,21 @@ func (n *insertNode) BatchedNext(params runParams) (bool, error) {
 		if next, err := n.source.Next(params); !next {
 			lastBatch = true
 			if err != nil {
+				// TODO(richardjcai): Don't like this, not sure how to check if the
+				// parse error is specifically from the column undergoing the
+				// alter column type schema change.
+
+				// Intercept parse error due to ALTER COLUMN TYPE schema change.
```

Comment:
Yeah, I don't think we can safely do this here... Not sure how to catch the error here.

Reply:
Jordan said it should be okay to say one of the columns is undergoing alter column type and say it MAY be because of this. The argument is that the user should expect behaviour like this for the online alter column type schema change.

Reply:
OK, that seems reasonable.

We should use the method that intercepts the error here though.